sgkit/stats/aggregation.py

```diff
@@ -113,17 +119,24 @@ def count_call_alleles(ds: Dataset, merge: bool = True) -> Dataset:
             )
         }
     )
-    return conditional_merge_datasets(ds, new_ds, merge)
+    return variables.validate(
+        conditional_merge_datasets(ds, new_ds, merge), "call_allele_count"
```

Comment:
@tomwhite this is outdated via 47eef29

Reply:
OK, but the question still remains where the dataset is created. A typo there won't cause validation to fail will it?

Reply:
@tomwhite a typo there would definitely cause a validation error (which is intentional/good). Whether we should reference variable or use strings is a good question, but more from a "dev practices" standpoint. I guess I lean towards referencing it, BUT it would mean that you need to for example say: `variables.pc_relate_phi.default_name` instead of `"pc_relate_phi"`. We could shorten it to `variables.pc_relate_phi`, if we provide some magic methods (hash/str). wdyt?

Reply:
Yes, I would probably lean towards referencing the variable too.

Reply:
for posterity, I've commented on the wrong comment section, copying over here:

> @tomwhite so I just remembered something, this would be going against some early comments in #17, what do you think? also ping @eric-czech

Reply:
I think using sgkit.variables for creating the output datasets sounds good.  The closest thing to an edge case I can think of right now is a function that generates a mask or mean imputes one of several possible input variables.  Perhaps validation could eventually be more dynamic for those cases.  Otherwise, I can't imagine where using a constant would be awkward.

Reply:
@tomwhite @eric-czech ok, this PR is now using `sgkit.variables` pretty much everywhere we need a variable name. I did not like the overly verbose `sgkit.variables.<varname>.default_name`, and making `Spec` hashable and acting like a string introduces complexity and will likely lead to issues in xr (see [here](https://github.com/ravwojdyla/sgkit/commit/b6197a94d155dba0099aa09a7030d9f2006bac20)), so I've opted the solution you can see in 6271e89.